src/searcher/glue.rs

```diff
@@ -75,33 +76,17 @@ impl<'s, S: Sink> MultiLine<'s, S> {
     pub fn run(mut self) -> Result<(), S::Error> {
         if self.core.begin()? {
             let mut keepgoing = true;
-            let mut query_cursor = QueryCursor::new();
-            let tree = get_parser(self.core.query_context().language)
-                .parse(self.slice, None)
-                .unwrap();
-            let query = self.core.query_context().query.clone();
-            let capture_index = self.core.query_context().capture_index;
-            let filter = self.core.query_context().filter.clone();
-            let mut matches = query_cursor
-                .captures(&query, tree.root_node(), self.slice)
-                .filter_map(|(match_, found_capture_index)| {
-                    let found_capture_index = found_capture_index as u32;
-                    if found_capture_index != capture_index {
-                        return None;
-                    }
-                    let mut nodes_for_this_capture = match_.nodes_for_capture_index(capture_index);
-                    let single_captured_node = nodes_for_this_capture.next().unwrap();
-                    assert!(
-                        nodes_for_this_capture.next().is_none(),
-                        "I guess .captures() always wraps up the single capture like this?"
-                    );
-                    match filter.as_ref() {
-                        None => Some(single_captured_node),
-                        Some(filter) => filter
-                            .call(&single_captured_node)
-                            .then_some(single_captured_node),
-                    }
-                });
+            let query_context = self.core.query_context();
+            let query = query_context.query.clone();
+            let filter = query_context.filter.clone();
+            let mut matches = get_captures(
```

Comment:
I did in fact DRY up the boilerplate of "running a tree-sitter query" and exposed those as `get_captures()`/`get_captures_for_enclosing_node()`